Xamarin.Essentials/Calendars/Calendars.uwp.cs

```diff
@@ -436,22 +436,39 @@ static async Task<string> PlatformCreateCalendar(Calendar newCalendar)
         static async Task<bool> PlatformDeleteCalendarEventInstanceByDate(string eventId, string calendarId, DateTimeOffset dateOfInstanceUtc)
         {
             await Permissions.RequestAsync<Permissions.CalendarWrite>();
+            var instance = await CalendarRequest.GetInstanceAsync(AppointmentStoreAccessType.AllCalendarsReadWrite);
 
             if (string.IsNullOrEmpty(eventId))
             {
                 throw new ArgumentException("[UWP]: You must supply an event id to delete an event.");
             }
-            var calendarEvent = await GetEventInstanceByIdAsync(eventId, dateOfInstanceUtc);
 
-            if (calendarEvent.CalendarId != calendarId)
+            Appointment uwpAppointment;
+            try
+            {
+                uwpAppointment = await instance.GetAppointmentInstanceAsync(eventId, dateOfInstanceUtc);
+            }
+            catch (ArgumentException)
+            {
+                if (string.IsNullOrWhiteSpace(eventId))
+                {
+                    throw new ArgumentException($"[UWP]: No Event found for event Id {eventId}");
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException($"[UWP]: No Event found for event Id {eventId}");
+                }
+            }
+
+            if (uwpAppointment.CalendarId != calendarId)
```

Comment:
shouldn't this if clause go inside the try-catch?

Reply:
I've moved it in, but it's a bit of a custom case, e.g. it won't be caught by the catch, either that or I can also make it throw an argument exception and then put the if logic within the catch, but then I feel like it'd be doubling up the if logic